packages/chat-stateful-client/src/ChatContext.ts

```diff
@@ -51,6 +53,16 @@ export class ChatContext {
     return this._state;
   }
 
+  public modifyState(modifier: ChatStateModifier): void {
```

Comment:
I'm curious why you chose an API where the developer passes a lambda instead of the modified state directly.

Reply:
The intent is to make writing a performant modifier easy (conversely, writing a bad one harder). This API allows the developer to "modify state in place". Thus, by default only necessary changes will be made. With an API that simply accepts an updated state, developer has to be more careful about how they modified the state (modify in place? Be careful about what parts are new vs not).

We expect all modifiers to use `immer` like facility. This API forces that choice on the developer - they're less likely to make mistakes and we can reason about how state is updated in a uniform way within the `Context` and via modifiers.

Reply:
That's what I suspected, we let the developer write a plain lambda without having to know how to use immer, sounds good!